### PR TITLE
Upgrade to node v20, 11ty v3, ESM throughout.

### PR DESCRIPTION
**Pull request by jgerigmeyer**

### CODE REVIEW

**Review by mirisuzanne — APPROVED** [review]
lgtm

File: content/_data/site.yaml [hunk 1]
```diff
@@ -11,11 +11,11 @@ styles:
...
-  email: birds
-  mastodon:
-    user: oddbird
-    server: front-end.social
-  github: oddbird
+  - email: birds
+  - mastodon:
+      user: oddbird
+      server: front-end.social
+  - github: oddbird
```
Comment (jgerigmeyer) [inline content/_data/site.yaml:18]: For reasons that aren't entirely clear to me, the order of keys in this dict was inconsistent before. Making it an array now gives us control over the order.

File: content/_includes/birds.macros.njk [hunk 2]
```diff
@@ -94,7 +94,7 @@ params:
     {%- set name = all or bird -%}
     {{ utility.link_if(
       content=name | mdInline | safe,
-      url=page.url if link else none,
+      url=page.page.url if link else none,
```
Comment (jgerigmeyer) [inline content/_includes/birds.macros.njk:97]: 11ty [recommends](https://www.11ty.dev/docs/collections/#collection-item-data-structure) accessing these props off the `page` object when in the context of a collection.

File: content/redirects.md [hunk 3]
```diff
@@ -13,5 +13,5 @@ permalink: /{{ redirect.from }}
 
 {% set to = collections.all | getPage(redirect.to) %}
 
-Redirecting you to [{{ post.name(to) }}]({{ to.url }}).
+Redirecting you to [{{ post.name(to) if to else redirect.to }}]({{ redirect.to }}).
```
Comment (jgerigmeyer) [inline content/redirects.md:16]: For reasons that aren't clear to me, at the time of this template compilation it seems that most of the `Tag` pages aren't yet part of `collections.all`. This is hardly visible in practice, but adds a fallback to display the URL if the page title isn't yet available.

**Review by jamesnw — APPROVED** [review]
This looks good! 

I did have some problems getting things set up- I think this was related to https://github.com/oddbird/oddleventy/pull/508 and using corepack. Do we need to add corepack directions to the README?

**Comment by jgerigmeyer** [comment]
> I did have some problems getting things set up- I think this was related to #508 and using corepack. Do we need to add corepack directions to the README?

Good reminder! Added in https://github.com/oddbird/oddleventy/pull/513/commits/6225477432b2e2268f3a5a0f0cc242344e5d1ee9

If others have problems using corepack, we can easily include Yarn in the repo like we did before.

**Comment by jgerigmeyer** [comment]
@mirisuzanne Re-requested your review of changes since your first review: https://github.com/oddbird/oddleventy/pull/513/files/9e12bb609fcc5a1ecb5c34a243dbae37efd8bc19..6225477432b2e2268f3a5a0f0cc242344e5d1ee9

**Review by mirisuzanne — APPROVED** [review]
lgtm

**Pull request merged**